### ISSUE TITLE
New Reader - Feature Requests & Thoughts

### ISSUE THREAD

**Issue by SeanLawler** [Tizra/Tizra-Customer-Tracker#76]
Hi David,

We really like the new reader that you're building and it seems really speedy.  However, there were a few issues that we noticed that we felt we should address as soon as possible seeing as how you're still working on it.

1) We will need alternative forward/back navigation for non-mobile users, or at least modify the current navigation it in such a way that it's intuitive for non-mobile users.  We think users will love the new reader, but they might not understand that they need to hover over the edge of the screen to display an arrow.

2) We will also need to be able to edit the reader's HTML.  To effectively brand this for our customers, we'll need to make changes to the header, footer, buttons, etc.

3) We'd like to be able to swap out the current Share This tool, with Omnipress' version, so we can take advantage of the metric tracking.

Thanks, and it's looking great!

-sml

**Comment by daviddurand** [comment]
Thanks for these thoughts. Although we used jQuery mobile widgets to implement it, this new version of the reader is actually targeted to desktop browsers. We wanted to pull the responsiveness of the mobile javascript into the desktop experience as an optimized, ready-to-go option. Once we've got that in place, we intend to see what design and functional ideas we should pull back into the mobile browser. The persistent sidebar is the major thing that's designed for the desktop, but would be terrible on the mobile platforms, with their smaller screen. I've tried it on the iPad and even on that large screen it makes the page view uncomfortably tiny.

The header area and color scheme are intended to be tailorable -- and that tailoring is one of the outstanding features I have to tie into the admin UI and the reader code. The buttons and scripts are not currently intended to be tailorable, as we're trying for a more consistent Tizra paging experience that we can engineer for usability.

The navigation buttons are modeled on the Google and Amazon readers (among others). The default now is slightly different than it was a few days ago. We now display the arrows on the initial presentation of a page, after which they fade and reappear based on mouse motion. We'll be continuing to test and look for feedback on this.

The share stuff is done using addthis, and getting it to work with  the ajax based navigation has been a bit tricky. Is there an account setup we can accomodate that could link your metrics via addthis? I didn't do that part of the scripting, so off the top of my head I'm not sure what the options are there; it just seems like there should be some.

If you are interested, we can look at extensions to enable additional customizations in any of these areas after the rollout. The new reader _is_ more locked-down, but using it will be a customer option, so the current ability to create custom reading pages will remain in place. I do think it might be hard to entirely replicate the new reader that way, as we made a few changes to the backend and the template logic to accomodate the ajax-based navigation, but I really haven't given it any thought at this time.

**Comment by orangedrink** [comment]
Our customers will definitely expect that we can make edits to this page and can make it consistent with other pages in the site (branding, header, footer, sidebar at a minimum)

We don't use addthis anymore because it was so slow and was causing freezes in IE. We're currently testing sharethis which seems faster but isn't able to be loaded asynchronously. We may not wind up using that in the end either.  

Have you checked out PDF.js? It can be used to render pdf files in browser with no plugins. All using javascript.
Here's a demo: http://mozilla.github.com/pdf.js/web/viewer.html
Maybe you know about it already but I thought it was worth mentioning.

**Comment by daviddurand** [comment]
We will be providing header and footer support, and basic style sheet tailoring for colors, etc.

The new reader is a tradeoff of slickness versus customizability, and that is why it will remain per-site option to use it in place of the existing, fully tailorable reading pages; so complex tailoring will remain an option, but will have to be done from scratch as it is today. The mobile platforms present so many constraints that they are where we have pushed the decision, and made the "iPad reader" the default.

AddThis is asynchronous, and it definitely has some quirks. Do make sure that you are using the most up-to-date AddThis code -- the old smart block code from our demos is definitely several versions old. Once we've got the basics in place, maybe we can make the social stuff a bit more tailorable.

pdf.js is very cool, but wasn't ready for primetime that last I saw it... Also, the range of crap that we get in PDF files makes me very wary of changing tools around, even when the current tools are not perfect. They are not perfect for a reason given the age of the PDF format, and the number of slightly buggy implementations of PDF creators and viewers. (they are all buggy, even Adobe's implementations). The jPedal guys whose code we use in the server have been in the PDF business for a long time, and they still find oddities that break rendering in particular PDF files. (and we've given them our share, of weird PDFs, courtesy of our publishers).
